src/main/java/bookstore/repository/book/spec/CoverImageSpecificationProvider.java

```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CoverImageSpecificationProvider implements SpecificationProvider<Book> {
```

Comment:
search by CoverImage? I can not imagine how it works

Reply:
But Google has such option (search by image) :) 
Ok, I've deleted it.